### PR TITLE
CDS connector addon

### PR DESCRIPTION
**Pull request by himanshu-sv**

### CODE REVIEW

File: addons/cds-connector-0.0.1/docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+type: Overview
+---
+
+Use **CDS Connector** to connect your CDS backend with Kyma events to trigger serverless compute and make API calls from a lambda or a microservice.
+
+**Prerequisite**
+
+It uses the API token for makinng API calls to CDS. Please provide the clientId and clientSecret in the form parameters field in following format 'client_id=id,grant_type=client_credentials,client_secret=secret'
+
+After adding this addon in the namespace, copy the http url of event-gw lambda  and provide it in the lambda running on CDS.
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/docs/overview.md]: Be aware of that the addons repo here is actually used. So even if declared as Preview we should assure that the documentation is good and not misleading.

Can you please improve the section here to be more specific. There will be a lambda deployed as part of the addon called "event-gw"? and I should copy the URL of the lambda? Please see also docs from other connectors as a reference. Feel free to ask @bszwarc for help on that

File: addons/cds-connector-0.0.1/chart/cds-connector/values.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Default values for github-connector.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+apiRegistrationJobName: api-registration-job
+oAuthProxyName: oauth-proxy-connector
+namespace: kyma-integration
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/chart/cds-connector/values.yaml]: The namespace is used for some resources, not for all. Is that really intended behavior? Shouldn't it all be located in the same namespace, for example the kyma-oauth-proxy deployment has no namespace specified and will end up in a different one.

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/oauth-proxy-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    metadata:
+      labels:
+        app: {{ template "bundle.fullname" . }}
+        component: {{ template "bundle.fullname" . }}
+      annotations:
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+      - name: {{ template "bundle.fullname" . }}
+        image: "jcawley5/kyma-oauth-proxy"
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/chart/cds-connector/templates/oauth-proxy-deployment.yaml]: What image is that? Please use official images only like the ones coming from kyma-incubator/connector-tools. Others might to support you in maintaining the connector and need a chance to change things.

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/gw.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+---
+apiVersion: gateway.kyma-project.io/v1alpha2
+kind: Api
+metadata:
+  labels:
+    function: cds-events
+    example: cds-events
+  name: cds-events
+spec:
+  hostname: "cds-event-{{ .Release.Namespace }}.{{ .Values.cluster_domain}}"
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/chart/cds-connector/templates/gw.yaml]: You expose the lambda without any authentication? And the lambda is putting the payload blindly in the event bus?
I do not see the relation to the oauth-proxy, when does that come into the game?

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/cm-spec.yaml [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            "source": "https://www.sap.com/products/commerce-cloud-context-driven-services.html"
+          }
+        ]
+      }
+    }
+  headers.json: |
+    {
+        "Authorization" : {{ printf "token %s" .Values.token | quote }}
+    }
+kind: ConfigMap
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/chart/cds-connector/templates/cm-spec.yaml]: please move the metadata block to the beginning of the file, that will improve readability.

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/cm-config.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    app: {{ template "bundle.fullname" . }}
+    component: {{ template "bundle.fullname" . }}-{{ .Values.apiRegistrationJobName }}
+data:
+  SYSTEM_URL	: {{ .Values.system_url | quote }}
+  APPLICATION_NAME: {{ .Values.application_name | quote }}
+  PROVIDER_NAME: CDS
+  PRODUCT_NAME: CDS
+  APP_KIND: rest-with-apikey
+  API_DESCRIPTION: "APIs for SAP CDS"
+  API_SPECIFICATION_URL: https://raw.githubusercontent.com/jokafreak/kyma-addons/master/addons/cds-connector-0.0.1/docs/api-specfication.json
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/chart/cds-connector/templates/cm-config.yaml]: Might need to be adjusted to this addons URL?

File: addons/cds-connector-0.0.1/README.md [hunk 7]
```diff
@@ -0,0 +1,8 @@
+Use **Preview CDS Connector** to connect your CDS backend with Kyma events to trigger serverless compute and make API calls from a lambda or a microservice.
+It is not recommended for production use. 
+
+**Prerequisite**
+
+It uses the API token for makinng API calls to CDS.
```
Comment (a-thaler) [inline addons/cds-connector-0.0.1/README.md]: which one?

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/oauth-proxy-deployment.yaml (hunk 3 above)
Comment (jcawley5) [reply addons/cds-connector-0.0.1/chart/cds-connector/templates/oauth-proxy-deployment.yaml]: I can place the code for this image in the repo.

File: addons/cds-connector-0.0.1/chart/cds-connector/templates/gw.yaml (hunk 4 above)
Comment (himanshu-sv) [reply addons/cds-connector-0.0.1/chart/cds-connector/templates/gw.yaml]: 1. For ease of configurability and being in preview mode, the lambda is not protected at the moment, there is another lambda on the CDS side which is preparing the appropriate payload. In the later iterations, I can add it.
2. To get the token from CDS, the OAuth credentials needs to be passed in the request body but the gateway passes them as header and also doesn't allow access to the body field. 
For that, we have created a proxy that takes the params and calls the token endpoint. The gateway then calls this proxy for token.

**Pull request merged**